client/me/purchases/manage-purchase/test/purchase-management-buttons.js

```diff
@@ -0,0 +1,168 @@
+	is_cancelable: false,
+	can_explicit_renew: true,
+	can_disable_auto_renew: false,
+	can_reenable_auto_renewal: false,
+	iap_purchase_management_link: null,
+	is_iap_purchase: false,
+	is_locked: false,
+	is_refundable: false,
+	refund_period_in_days: 14,
+	is_renewable: true,
+	is_renewal: false,
+	has_private_registration: false,
+	refund_amount: 0,
+	refund_currency_symbol: '$',
+	refund_text: '$0',
+	refund_options: null,
+	total_refund_amount: 0,
+	total_refund_text: '$0',
+	check_dns: false,
+};
+
+describe( 'Purchase Management Buttons', () => {
+	it( 'renders a cancel button when auto-renew is ON', async () => {
+		nock( 'https://public-api.wordpress.com' )
+			.get( '/rest/v1.1/me/payment-methods?expired=include' )
+			.reply( 200 );
+
+		const store = createReduxStore(
+			{
+				currentUser: { id: purchase.user_id },
+				happychat: { chat: { status: '' } },
+				plans: { items: [] },
+
+				purchases: {
+					data: [ { ...purchase, auto_renew: 1 } ],
+					hasLoadedUserPurchasesFromServer: true,
+					hasLoadedSitePurchasesFromServer: true,
+				},
+				productsList: { items: {} },
+				sites: {
+					plans: {},
+					requesting: {},
+					domains: { requesting: {}, items: { [ purchase.site_id ]: {} } },
+				},
+				plugins: {
+					premium: { plugins: {} },
+				},
+				ui: { selectSiteId: purchase.site_id },
+			},
+			( state ) => {
+				return state;
+			}
+		);
+
+		render(
+			<ReduxProvider store={ store }>
+				<ManagePurchase
+					purchaseId={ Number( purchase.ID ) }
+					isSiteLevel
+					siteSlug="onecooltestsite.com"
+				/>
+			</ReduxProvider>
+		);
+		expect( await screen.findByText( /Cancel/ ) ).toBeInTheDocument();
+	} );
+
+	it( 'renders a remove button when auto-renew is OFF', async () => {
+		nock( 'https://public-api.wordpress.com' )
+			.get( '/rest/v1.1/me/payment-methods?expired=include' )
+			.reply( 200 );
+
+		const store = createReduxStore(
+			{
+				currentUser: { id: purchase.user_id },
+				happychat: { chat: { status: '' } },
+				plans: { items: [] },
+
+				purchases: {
+					data: [ { ...purchase, auto_renew: 1 } ],
+					hasLoadedUserPurchasesFromServer: true,
+					hasLoadedSitePurchasesFromServer: true,
+				},
+				productsList: { items: {} },
+				sites: {
+					plans: {},
+					requesting: {},
+					domains: { requesting: {}, items: { [ purchase.site_id ]: {} } },
+				},
+				plugins: {
+					premium: { plugins: {} },
+				},
+				ui: { selectSiteId: purchase.site_id },
+			},
+			( state ) => {
+				return state;
+			}
+		);
+
+		render(
+			<ReduxProvider store={ store }>
+				<ManagePurchase
+					purchaseId={ Number( purchase.ID ) }
+					isSiteLevel
+					siteSlug="onecooltestsite.com"
+				/>
+			</ReduxProvider>
+		);
+		expect( await screen.findByText( /Remove/ ) ).toBeInTheDocument();
+	} );
+
+	// eslint-disable-next-line jest/no-commented-out-tests
+	// it.skip( 'renders a cancel button and does NOT render a remove button', () => {} );
```

Comment:
Probably better to remove these before this is merged, but maybe you had planned on finishing them before this was ready for review?

Reply:
Yep, these tests aren't finished yet. I was just looking for an initial once over on the changes to the purchase management page itself. TLDR this won't be merged until the tests are finished and passing.